ledger/nomos-ledger/src/mantle/sdp/rewards.rs

```diff
@@ -294,11 +325,54 @@ pub enum Error {
     InvalidProofType,
 }
 
+/// Creates a deterministic transaction hash for reward distribution.
+///
+/// The hash is computed from a version constant, session number, and service
+/// type, ensuring all nodes produce identical transaction hashes for reward
+/// notes.
+fn create_reward_tx_hash(session_n: SessionNumber, service_type: ServiceType) -> TxHash {
+    let mut hasher = ZkHasher::default();
+    let session_fr = Fr::from(session_n);
+    let service_type_fr = fr_from_bytes(service_type.as_ref().as_bytes())
+        .expect("Valid service type fr representation");
+    <ZkHasher as ZkDigest>::update(&mut hasher, &session_fr);
+    <ZkHasher as ZkDigest>::update(&mut hasher, &service_type_fr);
+
+    TxHash(hasher.finalize())
+}
```

Comment:
I think you have the order backwards here:
<img width="787" height="129" alt="Image" src="https://github.com/user-attachments/assets/fe880452-a286-4f37-8ae4-9824d076a1d0" />

Should be
```
    <ZkHasher as ZkDigest>::update(&mut hasher, &service_type_fr);
    <ZkHasher as ZkDigest>::update(&mut hasher, &session_fr);
```

Reply:
🦅